lua/scratch/config.lua

```diff
@@ -48,207 +56,80 @@ local default_config = {
-  -- Check if path contains any invalid characters
-  local valid_chars = "[^%w/%.%-%_]+"
-  if vim.fn.has("win32") == 1 then
-    valid_chars = "[^%w%.:%\\%-%_]+"
-  end
-  if string.match(path, valid_chars) then
-    return false
+---@return Scratch.Config
+local function get_and_update_config()
+  if not vim.g.scratch_json_config_path then
+    utils.log_err("Unable to locate json_config, please restart neovim and try again")
   end
 
-  -- Check if path ends with a forward slash
-  if string.sub(path, -1) == slash then
-    return false
-  end
+  local ok, json_config =
+    pcall(require("scratch.json").read_json_file, vim.g.scratch_json_config_path)
+  if not ok then
+    local msg = "Can't read the json config at: "
+      .. vim.g.scratch_json_config_path
+      .. ", please check the config or just delete it"
 
-  -- Check if path contains any double forward slashes
-  if string.match(path, slash .. slash) then
-    return false
+    utils.log_err(msg)
+    error(msg)
   end
 
-  -- if already exist check if it's a file
-  if vim.fn.filereadable(path) == 1 then
-    if vim.fn.isdirectory(path) == 1 then
-      return false
-    end
-  end
-
-  -- If all checks pass, return true
-  return true
-end
-
-M.checkInit = function()
-  return vim.fn.filereadable(CONFIG_FILE_FLAG_PATH) == 1 and validate_abspath(getConfigFilePath())
+  vim.g.scratch_config = json_config
+  return vim.g.scratch_config
 end
 
----Init the plugin
----@param force boolean
-local function initProcess(force)
-  -- if CONFIG_FILE_PATH file exist, don't need init
-  if vim.fn.filereadable(CONFIG_FILE_FLAG_PATH) == 1 and force ~= true then
-    if validate_abspath(getConfigFilePath()) then
-      return
-    else
-      logErr("Invalid path. Please rm `" .. CONFIG_FILE_FLAG_PATH .. "`and try again")
-    end
-  end
+local default_json_config_path = vim.fn.stdpath("config") .. slash .. "scratch_config.json" -- where the json config will be put
+vim.g.scratch_config = default_config
+vim.g.scratch_json_config_path = default_json_config_path
 
-  -- ask user to input the abspath of scratch file dir
-  local configFilePath =
-    vim.fn.input("Where you want to put your configuration file (abspath): ", DEFAULT_CONFIG_PATH)
-  if validate_abspath(configFilePath) == false then
-    vim.notify("invalid path. Path must be abspath and must be file type")
-    return
-  end
+---@param user_config? Scratch.SetupConfig
+local function setup(user_config)
+  user_config = user_config or {}
 
-  -- write the scratch_file_dir into CONFIG_FILE_PATH file
-  local dir_path = vim.fn.fnamemodify(CONFIG_FILE_FLAG_PATH, ":h")
-  if vim.fn.isdirectory(dir_path) == 0 then
-    vim.fn.mkdir(dir_path, "p")
+  local json_config_path = user_config and user_config.json_config_path or default_json_config_path
+  local json_config
+  if json.file_exists(json_config_path) then
+    json_config = json.read_json_file(json_config_path)
+  else
+    json.write_json_file(default_config, json_config_path)
```

Comment:
Why write `default_config`?

If the user wants to use `json_config` they can just save the file themselves.

Reply:
IMHO, if you want to make things easier for the user then having another file is already a poor choice)
It is usually best to keep things that belong together - close together.

I would recommend following the standard approach of how other plugins do it, i.e. just use `require("scratch").setup(opts)`.

For the instructions a good readme is usually better.